src/processors/BaseProcessor.cpp

```diff
@@ -356,20 +366,14 @@ juce::Point<int> BaseProcessor::getPosition (Rectangle<int> parentBounds)
     return (editorPosition * juce::Point { (float) parentBounds.getWidth(), (float) parentBounds.getHeight() }).toInt();
 }
 
-void BaseProcessor::routeExternalModulation (const std::initializer_list<int>& inputPorts, const std::initializer_list<int>& outputPorts)
-{
-    inputModulationPorts = inputPorts;
-    outputModulationPorts = outputPorts;
-}
-
 bool BaseProcessor::isInputModulationPort (int portIndex)
 {
-    return inputModulationPorts.contains (portIndex);
+    return inputPortTypes[portIndex] == PortType::modulation;
 }
 
 bool BaseProcessor::isOutputModulationPort (int portIndex)
 {
-    return outputModulationPorts.contains (portIndex);
+    return outputPortTypes[portIndex] == PortType::modulation;
 }
```

Comment:
Are we still using these methods? Since we now have `getInputPortType()` and `getOutputPortType()`, maybe we can remove these methods.

Reply:
`isInputModulationPort` is only used in BadModulationTest.cpp which doesn't sound like something that's being used 😄 

`isOutputModulationPort` is being used in `isOutputModulationPortConnected` if we still want to keep `isOutputModulationPortConnected` I can change the lines: 

`if (isOutputModulationPort (info.startPort))`
 `return true;`
to
`if (getOutputPortType(info.startPort) == PortType::modulation)` 
                `return true;`

Or do we want the connection checks to be generalised for multiple port types?

Reply:
Yeah, if we could switch to using the new methods in both places, that would be great!